internal/coreinternal/processor/filterconfig/config.go

```diff
@@ -107,6 +115,14 @@ type MatchProperties struct {
 	// A match occurs if the span's implementation library matches at least one item in this list.
 	// This is an optional field.
 	Libraries []InstrumentationLibrary `mapstructure:"libraries"`
+
+	// Expressions specifies the list of expr expressions to match metrics against.
+	// A match occurs if any datapoint in a metric matches at least one expression in this list.
+	Expressions []string `mapstructure:"expressions"`
```

Comment:
+1 to that, also if this functionality is really needed probably it should be extracted into separate PR (so one adds metrics, the other for expressions)

Reply:
I added expressions as a possible matching functionality as the metric filter already has the expression matching implemented. I assumed since it was a simple change it made sense to enable it.

From your feedback it looks like it adds more confusion to the customer than any benefit it would add, so I'll remove it for now (since there isn't a specific requirement to support expression matching at this point, AFAIK).